File: js/creditCheck.test.js
// Can you translate this driver code to unit tests?

var {creditCheck, sumDigits} = require("./creditCheck");

test('returns a string', () => {
  expect(typeof creditCheck("5541808923795240")).toBe('string')
})

test('valid numbers return "The number is valid!', () => {
  expect(creditCheck("5541808923795240")).toBe("The number is valid!")
  expect(creditCheck("4024007136512380")).toBe("The number is valid!")
  expect(creditCheck("6011797668867828")).toBe("The number is valid!")
})

test('invalid numbers return "The number is invalid!', () => {
  expect(creditCheck("5541801923795240")).toBe("The number is invalid!")
  expect(creditCheck("4024007106512380")).toBe("The number is invalid!")
  expect(creditCheck("6011797668868728")).toBe("The number is invalid!")
})

test('sum digits returns 4 for 13', () => {
  expect(sumDigits(13)).toBe(4)
})

File: python/credit_check_test.py
""" tests for credit check"""
from credit_check import credit_check, sum_digits


def test_returns_a_string():
    assert type(credit_check("4024007136512380")) == str


def test_returns_valid_for_valid_numbers():
    assert credit_check("5541808923795240") == "The number is valid!"
    assert credit_check("4024007136512380") == "The number is valid!"
    assert credit_check("6011797668867828") == "The number is valid!"


def test_returns_invalid_for_invalid_numbers():
    assert credit_check("5541801923795240") == "The number is invalid!"
    assert credit_check("4024007106512380") == "The number is invalid!"
    assert credit_check("6011797668868728") == "The number is invalid!"


def test_returns_4_for_13():
    assert sum_digits(13) == 4
